thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/anomaly/events/HolidayEventsLoader.java

```diff
@@ -0,0 +1,337 @@
+      this.endTime = endTime;
+    }
+
+    /**
+     * Gets event type.
+     *
+     * @return the event type
+     */
+    public String getEventType() {
+      return eventType;
+    }
+
+    /**
+     * Sets event type.
+     *
+     * @param eventType the event type
+     */
+    public void setEventType(String eventType) {
+      this.eventType = eventType;
+    }
+
+    @Override
+    public int hashCode() {
+      return Objects.hash(getName(), getEventType(), getStartTime(), getEndTime());
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+      if (!(obj instanceof HolidayEvent)) {
+        return false;
+      }
+      HolidayEvent holidayEvent = (HolidayEvent) obj;
+      return Objects.equals(getName(), holidayEvent.getName()) && Objects.equals(getStartTime(),
+          holidayEvent.getStartTime()) && Objects.equals(getEndTime(), holidayEvent.getEndTime()) && Objects.equals(
+          getEventType(), holidayEvent.getEventType());
+    }
+  }
+
+  /**
+   * Instantiates a new Holiday events loader.
+   *
+   * @param configuration the configuration
+   */
+  public HolidayEventsLoader(ThirdEyeAnomalyConfiguration configuration) {
+    calendarList = configuration.getCalendars();
+    keyPath = configuration.getRootDir() + configuration.getKeyPath();
+    scheduledExecutorService = Executors.newSingleThreadScheduledExecutor();
+  }
+
+  /** List of google holiday calendar ids */
+  private static List<String> calendarList;
+
+  /** Api private key path */
+  private static String keyPath;
+  private ScheduledExecutorService scheduledExecutorService;
+  private TimeGranularity runFrequency = new TimeGranularity(7, TimeUnit.DAYS);
+
+  private static final Logger LOG = LoggerFactory.getLogger(HolidayEventsLoader.class);
+
+  /** Global instance of the HTTP transport. */
+  private static HttpTransport HTTP_TRANSPORT;
+
+  /** Global instance of the JSON factory. */
+  private static final JsonFactory JSON_FACTORY = JacksonFactory.getDefaultInstance();
+
+  /** Global instance of the scopes. */
+  private static final Set<String> SCOPES = Collections.singleton(CalendarScopes.CALENDAR_READONLY);
+
+  static {
+    try {
+      HTTP_TRANSPORT = GoogleNetHttpTransport.newTrustedTransport();
+    } catch (Exception e) {
+      LOG.error("Can't create http transport with google api.", e);
+    }
+  }
+
+  private static final DAORegistry DAO_REGISTRY = DAORegistry.getInstance();
+
+  /**
+   * Start.
+   */
+  public void start() {
+    scheduledExecutorService.scheduleAtFixedRate(this, 0, runFrequency.getSize(), runFrequency.getUnit());
+  }
+
+  /**
+   * Shutdown.
+   */
+  public void shutdown() {
+    scheduledExecutorService.shutdown();
+  }
+
+  /**
+   * Fetch holidays and save to ThirdEye database.
+   */
+  public void run() {
+    EventManager eventDAO = DAO_REGISTRY.getEventDAO();
+
+    long start = System.currentTimeMillis();
+    long end = start + 2592000000L; // to get holidays within a month
```

Comment:
Store the value 2592000000L in the configuration file; like what you did for the key path and calendar list.